service/src/test/java/bio/terra/catalog/service/DatasetServiceTest.java

```diff
@@ -193,29 +180,36 @@ void testGetMetadataUsingTdrPermission() {
   void testUpdateMetadataWithInvalidUser() {
     mockDataset();
     assertThrows(
-        UnauthorizedException.class, () -> datasetService.updateMetadata(user, datasetId, "test"));
+        UnauthorizedException.class,
+        () -> datasetService.updateMetadata(user, datasetId, metadata));
   }
 
   @Test
   void testUpdateMetadata() {
     mockDataset();
-    String metadata = "test metadata";
     when(samService.hasGlobalAction(user, SamAction.UPDATE_ANY_METADATA)).thenReturn(true);
     datasetService.updateMetadata(user, datasetId, metadata);
     verify(datasetDao).update(dataset.withMetadata(metadata));
   }
 
+  @Test
+  void testUpdateMetadataInvalidInput() {
+    String metadata = "metadata must be json object";
```

Comment:
In general it's not good to have a variable shadow another variable in an outer scope. To avoid this, maybe use a different name for the local variable, like `badMetadata` or `notJson`.